### PR TITLE
feat(cli): Add local developer preview

### PR DESCRIPTION
**Pull request by tchock**

### CODE REVIEW

File: packages/cli/src/steps/downloadPreviewClient/downloadAndWriteClient.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { ZIP_PATH } from './constants';
+
+const fs = getFS();
+
+export async function downloadAndWriteClient(url): Promise<string> {
+  const writer = fs.createWriteStream(ZIP_PATH);
+  const res = await axios.get(url, {
+    responseType: 'stream',
+  });
+  res.data.pipe(writer);
```
Comment (JohannesMerz) [inline packages/cli/src/steps/downloadPreviewClient/downloadAndWriteClient.ts:13]: i guess its never an issue as the http request takes a tick, but you should always wait for the open event of the writer stream before piping

```
writer.on('open', () => res.data.pipe(writer));
```

Comment (JohannesMerz) [reply packages/cli/src/steps/downloadPreviewClient/downloadAndWriteClient.ts:13]: you might need to move waiting for the http request in front of 

```
const writer = fs.createWriteStream(ZIP_PATH);
```

in order to ad the event listner of the wirte stream in the same process.tick

File: packages/cli/src/commands/preview.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  .option('-d, --dir [dir]', 'The root folder to search components in')
+  .option(
+    '--webpack-config [path]',
+    'Path to the webpack config file, defaults to webpack.config.js'
+  )
+  .option('--port [port]', 'Port that the preview server is going to be available in')
+  .description('starts a local preview server')
+  .action(({ port }) => {
+    render(
+      <ConfigProvider config={{ dryRun: true, previewPort: port }}>
```
Comment (JohannesMerz) [inline packages/cli/src/commands/preview.tsx:19]: is dryRun: true intended here? i thought that was a cli option as well

Comment (tchock) [reply packages/cli/src/commands/preview.tsx:19]: Not yet. It's an internal config for now (only really needed for preview).

Comment (tchock) [reply packages/cli/src/commands/preview.tsx:19]: I guess it can be set through the bojagirc file, but that makes no sense at all :D

**Pull request merged**